Google.PowerShell/BigQuery/GcbqTable.cs

```diff
@@ -112,4 +112,179 @@ protected override void ProcessRecord()
+        [ValidatePattern("[a-zA-Z0-9_]")]
+        [ValidateLength(1, 1024)]
+        public string TableId { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// User-friendly name for the table.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.ByValues)]
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.ByValueWithRef)]
+        public string Name { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// Description of the table.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.ByValues)]
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.ByValueWithRef)]
+        public string Description { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The lifetime of this table from the time of creation (in seconds).
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.ByValues)]
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.ByValueWithRef)]
+        [ValidateRange(1, (long.MaxValue / 1000))]
+        public long Expiration { get; set; }
+
+        protected override void ProcessRecord()
+        {
+            // Set up the Dataset based on parameters
+            TablesResource.InsertRequest request;
+            switch (ParameterSetName)
+            {
+                case ParameterSetNames.ByObject:
+                    request = Service.Tables.Insert(InputObject,
+                        InputObject.TableReference.ProjectId,
+                        InputObject.TableReference.DatasetId);
+                    break;
+                case ParameterSetNames.ByValueWithRef:
+                    Project = Dataset.ProjectId;
+                    DatasetId = Dataset.DatasetId;
+                    request = makeInsertReq();
+                    break;
+                case ParameterSetNames.ByValues:
+                    request = makeInsertReq();
+                    break;
+                default:
+                    throw UnknownParameterSetException;
+            }
+
+            // Add the new dataset to the project supplied.
+            try
+            {
+                Table response = request.Execute();
+                WriteObject(response);
+            }
+            catch (GoogleApiException ex) when (ex.HttpStatusCode == HttpStatusCode.Conflict)
+            {
+                WriteError(new ErrorRecord(ex,
+                    $"A table with the name {TableId} already exists in {Project}:{DatasetId}.",
+                    ErrorCategory.InvalidArgument,
+                    TableId));
+            }
+        }
+
+        public TablesResource.InsertRequest makeInsertReq()
+        {
+            Table newTab;
```

Comment:
There is no need to split declaration and assignment here. The variable name should be the full word newTable.